src/vivarium_inputs/validation/raw.py

```
@@ -867,3 +876,25 @@ def check_cause_age_restrictions(entity: Cause) -> None:
             raise NotImplementedError(f'{entity.name} has a broader yll age range than yld age range.'
                                       f' We currently do not support these causes.')
 
+
+def check_paf_rr_age_groups(paf: pd.DataFrame, rr: pd.DataFrame) -> None:
+    """Raise an error if relative risk includes any age_group_ids with non
+    trivial data. But paf still can have an extra data while there is no
+    relative risk matching with.
+    """
+    measure_map = {MEASURES['YLLs']: 'YLLs', MEASURES['YLDs']: 'YLDs'}
+    cid = paf.cause_id.unique()[0]
+    measure = paf.measure_id.unique()[0]
+
+    if measure == MEASURES['YLLs']:
+        rr_data = rr[(rr.cause_id == cid) & (rr.morbidity == 0) & (rr.mortality == 1)]
+    else:
+        rr_data = rr[(rr.cause_id == cid) & (rr.morbidity == 1)]
+    missing_age_ids = set(rr_data.age_group_id)-set(paf.age_group_id)
```

So a few things that concern me about this test:

It depends on the rr and paf data representing missingness in the same way:

I remember that most of the exposure/rr data we pull uses true missing rows (rather than filled with zeros/ones) to represent no possible exposure in the data.  Do we know if that's universally true though?  

How does the paf data represent missingness?  I would think, since it's a final gbd output, that it comes with full age ranges with zero values set in appropriate places.

If either the rr data is inconsistent or the paf data represents missingness with zeros rather than deleted rows, doing a check on the age groups is not going to work correctly.  

The by age value checks we want are:
1. if there is nonzero exposure to the risk in an age bin and there are non-trivial rr values for the correct measure associated with that exposure, then there must be a nonzero paf, so raise an error
2. If there are nonzero paf values for age bins with zero exposure (or non-existent rows) or there are nonzero paf values for age bins with exposure but no rr values for the correct measure, then warn.  This is a situation where some age bins of paf were calculated directly, extrapolated, or something else we don't understand.

---

On the other side of this is cause restrictions.  If there are nonzero paf values for YLDs for a cause outside the cause yld restrictions, then we should raise an error.  Same for YLLs.